File: frontend/src/types/report.ts
/**
 * Report Types and Interfaces
 *
 * TypeScript type definitions for reporting and analytics data structures,
 * matching the backend Rust models for the Reporting & Analytics module (Milestone 12).
 */

// ========== DATE RANGE ==========

/**
 * Date range for filtering reports
 */
export interface DateRangeFilter {
  /** Start date (inclusive) - ISO 8601 format */
  start_date: string;
  /** End date (inclusive) - ISO 8601 format */
  end_date: string;
}

/**
 * Date range included in report responses
 */
export interface ReportDateRange {
  start_date: string;
  end_date: string;
}

// ========== APPOINTMENT REPORTS ==========

/**
 * Appointment report query parameters
 */
export interface AppointmentReportFilter {
  /** Start date (inclusive) */
  start_date?: string;
  /** End date (inclusive) */
  end_date?: string;
  /** Provider ID filter (optional) */
  provider_id?: string;
}

/**
 * Count by day of week
 */
export interface DayOfWeekCount {
  /** Day index (0=Sunday, 1=Monday, etc.) */
  day: number;
  /** Day name for display */
  day_name: string;
  /** Appointment count */
  count: number;
}

/**
 * Count by hour of day
 */
export interface HourlyCount {
  /** Hour (0-23) */
  hour: number;
  /** Appointment count */
  count: number;
}

/**
 * Daily appointment count for trend charts
 */
export interface DailyAppointmentCount {
  date: string;
  scheduled: number;
  completed: number;
  cancelled: number;
  no_shows: number;
}

/**
 * Appointment utilization report response
 */
export interface AppointmentUtilizationReport {
  /** Report date range */
  date_range: ReportDateRange;
  /** Total scheduled appointments in period */
  total_scheduled: number;
  /** Completed appointments */
  completed: number;
  /** Cancelled appointments */
  cancelled: number;
  /** No-show appointments */
  no_shows: number;
  /** Utilization rate (completed / scheduled * 100) */
  utilization_rate: number;
  /** No-show rate percentage */
  no_show_rate: number;
  /** Cancellation rate percentage */
  cancellation_rate: number;
  /** Average appointments per day */
  avg_appointments_per_day: number;
  /** Breakdown by appointment type */
  by_type: Record<string, number>;
  /** Breakdown by day of week (0=Sunday, 6=Saturday) */
  by_day_of_week: DayOfWeekCount[];
  /** Breakdown by hour of day */
  by_hour: HourlyCount[];
  /** Daily trend data */
  daily_trend: DailyAppointmentCount[];
}

// ========== PATIENT REPORTS ==========

/**
 * Patient statistics report query parameters
 */
export interface PatientReportFilter {
  /** Start date for new patient registration (optional) */
  start_date?: string;
  /** End date for new patient registration (optional) */
  end_date?: string;
}

/**
 * Gender breakdown statistics
 */
export interface GenderBreakdown {
  male: number;
  female: number;
  other: number;
  unspecified: number;
}

/**
 * Age group count
 */
export interface AgeGroupCount {
  /** Age group label (e.g., "0-18", "19-30", etc.) */
  age_group: string;
  /** Count of patients in group */
  count: number;
}

/**
 * Monthly count for trends
 */
export interface MonthlyCount {
  /** Year */
  year: number;
  /** Month (1-12) */
  month: number;
  /** Month name for display */
  month_name: string;
  /** Count */
  count: number;
}

/**
 * Patient statistics report response
 */
export interface PatientStatisticsReport {
  /** Report date range (if filtered) */
  date_range: ReportDateRange | null;
  /** Total patients in system */
  total_patients: number;
  /** Active patients */
  active_patients: number;
  /** Inactive patients */
  inactive_patients: number;
  /** Deceased patients */
  deceased_patients: number;
  /** Patients with active insurance */
  patients_with_insurance: number;
  /** New patients registered in period (if date filtered) */
  new_patients_in_period: number | null;
  /** Gender breakdown */
  by_gender: GenderBreakdown;
  /** Age distribution */
  age_distribution: AgeGroupCount[];
  /** Monthly registration trend (last 12 months) */
  monthly_registrations: MonthlyCount[];
}

// ========== DIAGNOSIS REPORTS ==========

/**
 * Diagnosis trends report query parameters
 */
export interface DiagnosisReportFilter {
  /** Start date (inclusive) */
  start_date?: string;
  /** End date (inclusive) */
  end_date?: string;
  /** Provider ID filter (optional) */
  provider_id?: string;
  /** Limit number of top diagnoses */
  limit?: number;
}

/**
 * Diagnosis frequency count
 */
export interface DiagnosisCount {
  /** ICD-10 code */
  icd10_code: string;
  /** Diagnosis description */
  description: string;
  /** Number of times diagnosed */
  count: number;
  /** Percentage of total */
  percentage: number;
}

/**
 * Monthly diagnosis count
 */
export interface MonthlyDiagnosisCount {
  year: number;
  month: number;
  month_name: string;
  count: number;
}

/**
 * Diagnosis category count (by ICD-10 chapter)
 */
export interface DiagnosisCategoryCount {
  /** Category code prefix (e.g., "A00-B99") */
  category: string;
  /** Category name */
  category_name: string;
  /** Count */
  count: number;
}

/**
 * Diagnosis trends report response
 */
export interface DiagnosisTrendsReport {
  /** Report date range */
  date_range: ReportDateRange | null;
  /** Total diagnoses recorded */
  total_diagnoses: number;
  /** Unique diagnosis codes */
  unique_codes: number;
  /** Top diagnoses by frequency */
  top_diagnoses: DiagnosisCount[];
  /** Monthly diagnosis trend */
  monthly_trend: MonthlyDiagnosisCount[];
  /** Diagnoses by category (ICD-10 chapter) */
  by_category: DiagnosisCategoryCount[];
}

// ========== PROVIDER PRODUCTIVITY REPORTS ==========

/**
 * Provider productivity report query parameters
 */
export interface ProductivityReportFilter {
  /** Start date (inclusive) */
  start_date?: string;
  /** End date (inclusive) */
  end_date?: string;
  /** Provider ID filter (optional, if not specified returns all providers) */
  provider_id?: string;
}

/**
 * Productivity summary
 */
export interface ProductivitySummary {
  /** Total appointments across all providers */
  total_appointments: number;
  /** Total completed appointments */
  completed_appointments: number;
  /** Total visits documented */
  total_visits: number;
  /** Total prescriptions written */
  total_prescriptions: number;
  /** Total documents generated */
  total_documents: number;
  /** Average appointment duration (minutes) */
  avg_appointment_duration: number;
}

/**
 * Individual provider productivity
 */
export interface ProviderProductivity {
  /** Provider user ID */
  provider_id: string;
  /** Provider name */
  provider_name: string;
  /** Provider role */
  provider_role: string;
  /** Appointments completed */
  appointments_completed: number;
  /** Visits documented */
  visits_documented: number;
  /** Prescriptions written */
  prescriptions_written: number;
  /** Documents generated */
  documents_generated: number;
  /** Patients seen (unique) */
  unique_patients_seen: number;
  /** Average visits per day */
  avg_visits_per_day: number;
  /** Completion rate */
  completion_rate: number;
}

/**
 * Provider productivity report response
 */
export interface ProviderProductivityReport {
  /** Report date range */
  date_range: ReportDateRange | null;
  /** Overall statistics */
  summary: ProductivitySummary;
  /** Per-provider breakdown (if multiple providers) */
  by_provider: ProviderProductivity[];
}

// ========== REVENUE REPORTS ==========

/**
 * Revenue report query parameters
 */
export interface RevenueReportFilter {
  /** Start date (inclusive) */
  start_date?: string;
  /** End date (inclusive) */
  end_date?: string;
  /** Provider ID filter (optional) */
  provider_id?: string;
}

/**
 * Revenue report response (placeholder - based on visits until billing integration)
 */
export interface RevenueReport {
  /** Report date range */
  date_range: ReportDateRange | null;
  /** Total visits in period */
  total_visits: number;
  /** Visits by type */
  visits_by_type: Record<string, number>;
  /** Average visits per day */
  avg_visits_per_day: number;
  /** Note about billing integration requirement */
  note: string;
}

// ========== DASHBOARD REPORT ==========

/**
 * Recent appointment summary
 */
export interface RecentAppointment {
  id: string;
  patient_name: string;
  scheduled_start: string;
  appointment_type: string;
  status: string;
}

/**
 * Recent visit summary
 */
export interface RecentVisit {
  id: string;
  patient_name: string;
  visit_date: string;
  visit_type: string;
  status: string;
}

/**
 * New patient summary
 */
export interface NewPatientSummary {
  id: string;
  name: string;
  registered_at: string;
}

/**
 * Quick statistics for dashboard
 */
export interface QuickStats {
  /** Today's appointments */
  appointments_today: number;
  /** Appointments this week */
  appointments_this_week: number;
  /** Pending visits to document */
  pending_visits: number;
  /** Active patients */
  active_patients: number;
  /** Documents generated this month */
  documents_this_month: number;
}

/**
 * Recent activity summary
 */
export interface RecentActivity {
  /** Recent appointments (last 5) */
  recent_appointments: RecentAppointment[];
  /** Recent visits (last 5) */
  recent_visits: RecentVisit[];
  /** New patients (last 5) */
  new_patients: NewPatientSummary[];
}

/**
 * Dashboard overview report
 */
export interface DashboardReport {
  /** Report generation timestamp */
  generated_at: string;
  /** Quick stats for dashboard widgets */
  quick_stats: QuickStats;
  /** Recent activity */
  recent_activity: RecentActivity;
}

// ========== EXPORT ==========

/**
 * Export format options
 */
export enum ExportFormat {
  JSON = 'json',
  CSV = 'csv',
  PDF = 'pdf',
  EXCEL = 'excel',
}

/**
 * Report type for export
 */
export enum ReportType {
  APPOINTMENT_UTILIZATION = 'appointment_utilization',
  PATIENT_STATISTICS = 'patient_statistics',
  DIAGNOSIS_TRENDS = 'diagnosis_trends',
  PROVIDER_PRODUCTIVITY = 'provider_productivity',
  REVENUE = 'revenue',
  DASHBOARD = 'dashboard',
}

/**
 * Export request
 */
export interface ExportReportRequest {
  /** Report type to export */
  report_type: ReportType;
  /** Export format */
  format: ExportFormat;
  /** Date range filter */
  start_date?: string;
  end_date?: string;
  /** Provider filter */
  provider_id?: string;
}

// ========== HELPER FUNCTIONS ==========

/**
 * Get color for report status badge
 * @param rate - Percentage rate value
 * @param thresholds - Optional custom thresholds [good, warning]
 * @returns Tailwind color class
 */
export function getRateColor(rate: number, thresholds = [80, 60]): string {
  if (rate >= thresholds[0]) {
    return 'text-green-600 dark:text-green-400';
  } else if (rate >= thresholds[1]) {
    return 'text-yellow-600 dark:text-yellow-400';
  }
  return 'text-red-600 dark:text-red-400';
}

/**
 * Format percentage for display
 * @param value - Percentage value
 * @param decimals - Number of decimal places
 * @returns Formatted percentage string
 */
export function formatPercentage(value: number, decimals = 1): string {
  return `${value.toFixed(decimals)}%`;
}

/**
 * Format number with thousands separator
 * @param value - Number to format
 * @returns Formatted number string
 */
export function formatNumber(value: number): string {
  return value.toLocaleString();
}

/**
 * Get chart colors for consistent styling
 * @returns Object with chart color values
 */
export function getChartColors() {
  return {
    primary: '#3B82F6', // blue-500
    success: '#10B981', // emerald-500
    warning: '#F59E0B', // amber-500
    danger: '#EF4444', // red-500
    info: '#8B5CF6', // violet-500
    muted: '#6B7280', // gray-500
    // Extended palette for multi-series charts
    palette: [
      '#3B82F6', // blue
      '#10B981', // emerald
      '#F59E0B', // amber
      '#EF4444', // red
      '#8B5CF6', // violet
      '#EC4899', // pink
      '#14B8A6', // teal
      '#F97316', // orange
    ],
  };
}

File: backend/src/models/report.rs
/*!
 * Report Models
 *
 * Data structures for reporting and analytics endpoints.
 * Supports date range filtering for all report types.
 */

use chrono::{DateTime, NaiveDate, Utc};
use serde::{Deserialize, Serialize};
use std::collections::HashMap;
use uuid::Uuid;

/// Date range filter for reports
#[derive(Debug, Clone, Deserialize)]
pub struct DateRangeFilter {
    /// Start date (inclusive)
    pub start_date: NaiveDate,
    /// End date (inclusive)
    pub end_date: NaiveDate,
}

impl DateRangeFilter {
    /// Create a new date range filter
    pub fn new(start_date: NaiveDate, end_date: NaiveDate) -> Self {
        Self {
            start_date,
            end_date,
        }
    }

    /// Validate that end date is not before start date
    pub fn validate(&self) -> Result<(), String> {
        if self.end_date < self.start_date {
            return Err("End date cannot be before start date".to_string());
        }
        Ok(())
    }
}

// ========== APPOINTMENT REPORTS ==========

/// Appointment utilization report query parameters
#[derive(Debug, Clone, Deserialize)]
pub struct AppointmentReportFilter {
    /// Start date (inclusive)
    pub start_date: Option<NaiveDate>,
    /// End date (inclusive)
    pub end_date: Option<NaiveDate>,
    /// Provider ID filter (optional)
    pub provider_id: Option<Uuid>,
}

/// Appointment utilization report response
#[derive(Debug, Clone, Serialize)]
pub struct AppointmentUtilizationReport {
    /// Report date range
    pub date_range: ReportDateRange,
    /// Total scheduled appointments in period
    pub total_scheduled: i64,
    /// Completed appointments
    pub completed: i64,
    /// Cancelled appointments
    pub cancelled: i64,
    /// No-show appointments
    pub no_shows: i64,
    /// Utilization rate (completed / scheduled * 100)
    pub utilization_rate: f64,
    /// No-show rate percentage
    pub no_show_rate: f64,
    /// Cancellation rate percentage
    pub cancellation_rate: f64,
    /// Average appointments per day
    pub avg_appointments_per_day: f64,
    /// Breakdown by appointment type
    pub by_type: HashMap<String, i64>,
    /// Breakdown by day of week (0=Sunday, 6=Saturday)
    pub by_day_of_week: Vec<DayOfWeekCount>,
    /// Breakdown by hour of day
    pub by_hour: Vec<HourlyCount>,
    /// Daily trend data
    pub daily_trend: Vec<DailyAppointmentCount>,
}

/// Date range included in report response
#[derive(Debug, Clone, Serialize)]
pub struct ReportDateRange {
    pub start_date: NaiveDate,
    pub end_date: NaiveDate,
}

/// Count by day of week
#[derive(Debug, Clone, Serialize)]
pub struct DayOfWeekCount {
    /// Day index (0=Sunday, 1=Monday, etc.)
    pub day: i32,
    /// Day name for display
    pub day_name: String,
    /// Appointment count
    pub count: i64,
}

/// Count by hour of day
#[derive(Debug, Clone, Serialize)]
pub struct HourlyCount {
    /// Hour (0-23)
    pub hour: i32,
    /// Appointment count
    pub count: i64,
}

/// Daily appointment count for trend charts
#[derive(Debug, Clone, Serialize)]
pub struct DailyAppointmentCount {
    pub date: NaiveDate,
    pub scheduled: i64,
    pub completed: i64,
    pub cancelled: i64,
    pub no_shows: i64,
}

// ========== PATIENT REPORTS ==========

/// Patient statistics report query parameters
#[derive(Debug, Clone, Deserialize)]
pub struct PatientReportFilter {
    /// Start date for new patient registration (optional)
    pub start_date: Option<NaiveDate>,
    /// End date for new patient registration (optional)
    pub end_date: Option<NaiveDate>,
}

/// Patient statistics report response
#[derive(Debug, Clone, Serialize)]
pub struct PatientStatisticsReport {
    /// Report date range (if filtered)
    pub date_range: Option<ReportDateRange>,
    /// Total patients in system
    pub total_patients: i64,
    /// Active patients
    pub active_patients: i64,
    /// Inactive patients
    pub inactive_patients: i64,
    /// Deceased patients
    pub deceased_patients: i64,
    /// Patients with active insurance
    pub patients_with_insurance: i64,
    /// New patients registered in period (if date filtered)
    pub new_patients_in_period: Option<i64>,
    /// Gender breakdown
    pub by_gender: GenderBreakdown,
    /// Age distribution
    pub age_distribution: Vec<AgeGroupCount>,
    /// Monthly registration trend (last 12 months)
    pub monthly_registrations: Vec<MonthlyCount>,
}

/// Gender breakdown statistics
#[derive(Debug, Clone, Serialize)]
pub struct GenderBreakdown {
    pub male: i64,
    pub female: i64,
    pub other: i64,
    pub unspecified: i64,
}

/// Age group count
#[derive(Debug, Clone, Serialize)]
pub struct AgeGroupCount {
    /// Age group label (e.g., "0-18", "19-30", etc.)
    pub age_group: String,
    /// Count of patients in group
    pub count: i64,
}

/// Monthly count for trends
#[derive(Debug, Clone, Serialize)]
pub struct MonthlyCount {
    /// Year
    pub year: i32,
    /// Month (1-12)
    pub month: i32,
    /// Month name for display
    pub month_name: String,
    /// Count
    pub count: i64,
}

// ========== DIAGNOSIS REPORTS ==========

/// Diagnosis trends report query parameters
#[derive(Debug, Clone, Deserialize)]
pub struct DiagnosisReportFilter {
    /// Start date (inclusive)
    pub start_date: Option<NaiveDate>,
    /// End date (inclusive)
    pub end_date: Option<NaiveDate>,
    /// Provider ID filter (optional)
    pub provider_id: Option<Uuid>,
    /// Limit number of top diagnoses
    pub limit: Option<i32>,
}

/// Diagnosis trends report response
#[derive(Debug, Clone, Serialize)]
pub struct DiagnosisTrendsReport {
    /// Report date range
    pub date_range: Option<ReportDateRange>,
    /// Total diagnoses recorded
    pub total_diagnoses: i64,
    /// Unique diagnosis codes
    pub unique_codes: i64,
    /// Top diagnoses by frequency
    pub top_diagnoses: Vec<DiagnosisCount>,
    /// Monthly diagnosis trend
    pub monthly_trend: Vec<MonthlyDiagnosisCount>,
    /// Diagnoses by category (ICD-10 chapter)
    pub by_category: Vec<DiagnosisCategoryCount>,
}

/// Diagnosis frequency count
#[derive(Debug, Clone, Serialize)]
pub struct DiagnosisCount {
    /// ICD-10 code
    pub icd10_code: String,
    /// Diagnosis description
    pub description: String,
    /// Number of times diagnosed
    pub count: i64,
    /// Percentage of total
    pub percentage: f64,
}

/// Monthly diagnosis count
#[derive(Debug, Clone, Serialize)]
pub struct MonthlyDiagnosisCount {
    pub year: i32,
    pub month: i32,
    pub month_name: String,
    pub count: i64,
}

/// Diagnosis category count (by ICD-10 chapter)
#[derive(Debug, Clone, Serialize)]
pub struct DiagnosisCategoryCount {
    /// Category code prefix (e.g., "A00-B99")
    pub category: String,
    /// Category name
    pub category_name: String,
    /// Count
    pub count: i64,
}

// ========== PROVIDER PRODUCTIVITY REPORTS ==========

/// Provider productivity report query parameters
#[derive(Debug, Clone, Deserialize)]
pub struct ProductivityReportFilter {
    /// Start date (inclusive)
    pub start_date: Option<NaiveDate>,
    /// End date (inclusive)
    pub end_date: Option<NaiveDate>,
    /// Provider ID filter (optional, if not specified returns all providers)
    pub provider_id: Option<Uuid>,
}

/// Provider productivity report response
#[derive(Debug, Clone, Serialize)]
pub struct ProviderProductivityReport {
    /// Report date range
    pub date_range: Option<ReportDateRange>,
    /// Overall statistics
    pub summary: ProductivitySummary,
    /// Per-provider breakdown (if multiple providers)
    pub by_provider: Vec<ProviderProductivity>,
}

/// Productivity summary
#[derive(Debug, Clone, Serialize)]
pub struct ProductivitySummary {
    /// Total appointments across all providers
    pub total_appointments: i64,
    /// Total completed appointments
    pub completed_appointments: i64,
    /// Total visits documented
    pub total_visits: i64,
    /// Total prescriptions written
    pub total_prescriptions: i64,
    /// Total documents generated
    pub total_documents: i64,
    /// Average appointment duration (minutes)
    pub avg_appointment_duration: f64,
}

/// Individual provider productivity
#[derive(Debug, Clone, Serialize)]
pub struct ProviderProductivity {
    /// Provider user ID
    pub provider_id: Uuid,
    /// Provider name
    pub provider_name: String,
    /// Provider role
    pub provider_role: String,
    /// Appointments completed
    pub appointments_completed: i64,
    /// Visits documented
    pub visits_documented: i64,
    /// Prescriptions written
    pub prescriptions_written: i64,
    /// Documents generated
    pub documents_generated: i64,
    /// Patients seen (unique)
    pub unique_patients_seen: i64,
    /// Average visits per day
    pub avg_visits_per_day: f64,
    /// Completion rate
    pub completion_rate: f64,
}

// ========== REVENUE REPORTS (Optional) ==========

/// Revenue report query parameters
#[derive(Debug, Clone, Deserialize)]
pub struct RevenueReportFilter {
    /// Start date (inclusive)
    pub start_date: Option<NaiveDate>,
    /// End date (inclusive)
    pub end_date: Option<NaiveDate>,
    /// Provider ID filter (optional)
    pub provider_id: Option<Uuid>,
}

/// Revenue report response (placeholder - can be expanded based on billing integration)
#[derive(Debug, Clone, Serialize)]
pub struct RevenueReport {
    /// Report date range
    pub date_range: Option<ReportDateRange>,
    /// Total visits in period
    pub total_visits: i64,
    /// Visits by type
    pub visits_by_type: HashMap<String, i64>,
    /// Average visits per day
    pub avg_visits_per_day: f64,
    /// Note: Actual revenue tracking requires billing module integration
    pub note: String,
}

// ========== COMBINED DASHBOARD REPORT ==========

/// Dashboard overview report
#[derive(Debug, Clone, Serialize)]
pub struct DashboardReport {
    /// Report generation timestamp
    pub generated_at: DateTime<Utc>,
    /// Quick stats for dashboard widgets
    pub quick_stats: QuickStats,
    /// Recent activity
    pub recent_activity: RecentActivity,
}

/// Quick statistics for dashboard
#[derive(Debug, Clone, Serialize)]
pub struct QuickStats {
    /// Today's appointments
    pub appointments_today: i64,
    /// Appointments this week
    pub appointments_this_week: i64,
    /// Pending visits to document
    pub pending_visits: i64,
    /// Active patients
    pub active_patients: i64,
    /// Documents generated this month
    pub documents_this_month: i64,
}

/// Recent activity summary
#[derive(Debug, Clone, Serialize)]
pub struct RecentActivity {
    /// Recent appointments (last 5)
    pub recent_appointments: Vec<RecentAppointment>,
    /// Recent visits (last 5)
    pub recent_visits: Vec<RecentVisit>,
    /// New patients (last 5)
    pub new_patients: Vec<NewPatientSummary>,
}

/// Recent appointment summary
#[derive(Debug, Clone, Serialize)]
pub struct RecentAppointment {
    pub id: Uuid,
    pub patient_name: String,
    pub scheduled_start: DateTime<Utc>,
    pub appointment_type: String,
    pub status: String,
}

/// Recent visit summary
#[derive(Debug, Clone, Serialize)]
pub struct RecentVisit {
    pub id: Uuid,
    pub patient_name: String,
    pub visit_date: NaiveDate,
    pub visit_type: String,
    pub status: String,
}

/// New patient summary
#[derive(Debug, Clone, Serialize)]
pub struct NewPatientSummary {
    pub id: Uuid,
    pub name: String,
    pub registered_at: DateTime<Utc>,
}

// ========== EXPORT FORMATS ==========

/// Export format options
#[derive(Debug, Clone, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum ExportFormat {
    Json,
    Csv,
    Pdf,
    Excel,
}

impl Default for ExportFormat {
    fn default() -> Self {
        Self::Json
    }
}

/// Report type for export
#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum ReportType {
    AppointmentUtilization,
    PatientStatistics,
    DiagnosisTrends,
    ProviderProductivity,
    Revenue,
    Dashboard,
}

/// Export request
#[derive(Debug, Clone, Deserialize)]
pub struct ExportReportRequest {
    /// Report type to export
    pub report_type: ReportType,
    /// Export format
    #[serde(default)]
    pub format: ExportFormat,
    /// Date range filter
    pub start_date: Option<NaiveDate>,
    pub end_date: Option<NaiveDate>,
    /// Provider filter
    pub provider_id: Option<Uuid>,
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_date_range_filter_validation() {
        let valid = DateRangeFilter::new(
            NaiveDate::from_ymd_opt(2024, 1, 1).unwrap(),
            NaiveDate::from_ymd_opt(2024, 12, 31).unwrap(),
        );
        assert!(valid.validate().is_ok());

        let invalid = DateRangeFilter::new(
            NaiveDate::from_ymd_opt(2024, 12, 31).unwrap(),
            NaiveDate::from_ymd_opt(2024, 1, 1).unwrap(),
        );
        assert!(invalid.validate().is_err());
    }

    #[test]
    fn test_export_format_default() {
        assert_eq!(ExportFormat::default(), ExportFormat::Json);
    }
}
